skyve-war/src/main/java/modules/admin/Tag/TagService.java

```diff
@@ -0,0 +1,144 @@
+package modules.admin.Tag;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.skyve.CORE;
+import org.skyve.EXT;
+import org.skyve.domain.Bean;
+import org.skyve.metadata.customer.Customer;
+import org.skyve.metadata.model.document.Document;
+import org.skyve.metadata.module.Module;
+import org.skyve.metadata.user.User;
+import org.skyve.persistence.AutoClosingIterable;
+import org.skyve.persistence.Persistence;
+import org.skyve.persistence.SQL;
+import org.skyve.tag.TagManager;
+
+import jakarta.enterprise.inject.Default;
+
+/**
+ * This class acts as a service layer to encapsulate domain logic.
+ *
+ * Add this line to classes that wish to use it: @Inject private transient TagService tagService;
+ */
+@Default
+public class TagService {
+	/**
+	 * Add to the subject tag records from the object tag.
+	 * 
+	 * @param subject
+	 * @param object
+	 */
+	@SuppressWarnings("static-method")
+	public void union(TagExtension subject, TagExtension object) throws Exception {
+		if (subject != null && object != null) {
+			// no dialect insensitive way to use SQL for creating new UUIDs for new records
+			// add tagged items from object tag to subject tag
+			// TagManager function deals with duplicates
+			TagManager tm = EXT.getTagManager();
+			try (AutoClosingIterable<Bean> i = tm.iterate(object.getBizId())) {
+				for (Bean bean : i) {
+					tm.tag(subject.getBizId(), bean);
+				}
+			}
+			subject.setUploadTagged(
+					Long.valueOf(subject.countDocument(subject.getUploadModuleName(), subject.getUploadDocumentName())));
+			subject.setTotalTagged(Long.valueOf(subject.count()));
+		}
+	}
+
+	/**
+	 * Deletes from the subject Tag any items which are not also in the object
+	 * Tag.
+	 * 
+	 * @param subject
+	 * @param object
+	 * @throws Exception
+	 */
+	@SuppressWarnings("static-method")
+	public void intersect(TagExtension subject, TagExtension object) throws Exception {
+		if (subject != null && object != null) {
+			Persistence pers = CORE.getPersistence();
+
+			// unsecured SQL for performance
+			StringBuilder intersect = new StringBuilder();
+			intersect.append("delete from ADM_Tagged ");
+			intersect.append(" where taggedBizId not in (");
+			intersect.append(" select taggedBizId from ADM_Tagged where ");
+			intersect.append(" tag_id = :objectTagId");
+			intersect.append(" and bizCustomer=:objectBizCustomer").append(" ) ");
+			intersect.append(" and tag_id =:subjectTagId");
+			intersect.append(" and bizCustomer=:subjectBizCustomer");
```

Comment:
Maybe use a multi-line String? Doesn't look like any variables are being added here that I can see.